client/elements/text/sc-bilara-segmented-text.js

```diff
@@ -384,18 +423,20 @@ class SCBilaraSegmentedText extends SCLitTextPage {
   _prepareNavigation() {
     let sutta = this.bilaraTranslatedSutta ? this.bilaraTranslatedSutta : this.bilaraRootSutta;
     if (!sutta) {
+      this.actions.showToc([]);
       return;
     }
     const dummyElement = document.createElement('template');
     dummyElement.innerHTML = this.markup.trim();
     let arrayTOC = Array.from(dummyElement.content.querySelectorAll('h2')).map(elem => {
       const id = elem.firstElementChild.id;
-      if (sutta.strings[id]) {
-        return { link: id, name: this._stripLeadingOrdering(sutta.strings[id]) };
+      if (sutta[id]) {
+        return { link: id, name: this._stripLeadingOrdering(sutta[id]) };
       }
     });
     arrayTOC = arrayTOC.filter(Boolean);
-    return arrayTOC;
+
+    this.actions.showToc(arrayTOC);
   }

```

Comment:
You should not add structure name in variable name, it should be an abstract name.